### PR TITLE
Add new E2 function `getUserGroup`

### PR DESCRIPTION
**Pull request by bjcscat**
Would be useful for getting user information beyond the IsAdmin or IsSuperAdmin functions provided.

### CODE REVIEW

**Comment by Aws0mee** [comment]
based

File: lua/entities/gmod_wire_expression2/core/player.lua [hunk 1]
```diff
@@ -21,6 +21,12 @@ end)
 
 __e2setcost(5) -- temporary
 
+e2function steam entity:getUserGroup()
```
Comment (Denneisk) [inline lua/entities/gmod_wire_expression2/core/player.lua:24]: This returns a *what*?

Comment (bjcscat) [reply lua/entities/gmod_wire_expression2/core/player.lua]: went braindead for a second i guess, fixed it

**Comment by Aws0mee** [comment]
new steam datatype just dropped 🔥

File: lua/entities/gmod_wire_expression2/core/player.lua [hunk 2]
```diff
@@ -21,6 +21,12 @@ end)
 
 __e2setcost(5) -- temporary
 
+e2function string entity:getUserGroup()
+	if not IsValid(this) then return self:throw("Invalid entity!", 0) end
+	if not this:IsPlayer() then return self:throw("Expected a Player but got an Entity!", 0) end
```
Comment (Denneisk) [inline lua/entities/gmod_wire_expression2/core/player.lua:26]: These should return empty strings instead of 0.

Comment (bjcscat) [reply lua/entities/gmod_wire_expression2/core/player.lua]: corrected

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**